assets/common/components/batch_deploy_model/spec.yaml

```diff
@@ -0,0 +1,191 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
+name: batch_deploy_model
+version: 0.0.1
+type: command
+
+is_deterministic: True
+
+display_name: Batch deploy model
+description:
+  Batch deploy a model to a workspace. The component works on compute with [MSI](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-create-manage-compute-instance?tabs=python) attached.
+
+environment: azureml://registries/azureml/environments/python-sdk-v2/versions/6
+
+code: ../../src
+command: >-
+  python batch_deploy.py
+  $[[--registration_details ${{inputs.registration_details}}]]
+  $[[--model_id ${{inputs.model_id}}]]
+  $[[--inference_payload ${{inputs.inference_payload}}]]
+  $[[--endpoint_name ${{inputs.endpoint_name}}]]
+  $[[--deployment_name ${{inputs.deployment_name}}]]
+  $[[--compute_name ${{inputs.compute_name}}]]
+  $[[--size ${{inputs.size}}]]
+  $[[--min_instances ${{inputs.min_instances}}]]
+  $[[--max_instances ${{inputs.max_instances}}]]
+  $[[--idle_time_before_scale_down ${{inputs.idle_time_before_scale_down}}]]
+  $[[--output_file_name ${{inputs.output_file_name}}]]
+  $[[--max_concurrency_per_instance ${{inputs.max_concurrency_per_instance}}]]
+  $[[--error_threshold ${{inputs.error_threshold}}]]
+  $[[--max_retries ${{inputs.max_retries}}]]
+  $[[--timeout ${{inputs.timeout}}]]
+  $[[--logging_level ${{inputs.logging_level}}]]
+  $[[--mini_batch_size ${{inputs.mini_batch_size}}]]
+  $[[--instance_count ${{inputs.instance_count}}]]
+  --model_deployment_details ${{outputs.model_deployment_details}}
+
+inputs:
+  # Output of registering component
+  registration_details:
+    type: uri_file
+    optional: true
+    description: Json file that contains the ID of registered model to be deployed
+
+  model_id:
+    type: string
+    optional: true 
+    description: |
+      Asset ID of the model registered in workspace/registry.
+      Registry - azureml://registries/<registry-name>/models/<model-name>/versions/<version>
+      Workspace - azureml:<model-name>:<version>
+
+  inference_payload:
+    type: uri_folder
+    optional: true
+    description: Folder containing files used to validate deployment
+
+  endpoint_name:
+    type: string
+    optional: true
+    description: Name of the endpoint
+
+  deployment_name:
+    type: string
+    optional: true
+    default: default
+    description: Name of the deployment
+    
+  compute_name:
```

Comment:
Could we add in description that if compute name is provided new compute will not be created?

Reply:
Added